File: lab 2/fib.java
import java.util.*;

class fib{
    static int fib(int n){
        if(n<=1){
            return n;
        }
        return fib(n-1)+fib(n-2);
    }
    static void printfib(int n){
        if (n < 0) {
            System.out.println("Please enter a positive integer.");
            return;
        }
        for(int i=0;i<n;i++){
            System.out.print(fib(i)+" ");
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of terms:");
        int n = sc.nextInt();
        printfib(n);
        sc.close();
    }
}

File: lab 2/fib.cpp
#include <iostream>
using namespace std;

int fib(int n)
{
    if (n == 0)
    {
        return 0;
    }
    if (n == 1)
    {
        return 1;
    }
    return fib(n - 1) + fib(n - 2);
}

void printfib(int n)
{
    if (n == 0 | n == 1)
    {
        cout << n << endl;
    }

    for (int i = 0; i < n; i++)
    {
        cout << fib(i) << " ";
    }
    cout << endl;
}

int main()
{
    int n;
    cout << "Enter the number of terms: ";
    cin >> n;
    if (n < 0)
    {
        cout << "Invalid input" << endl;
    }
    else
    {
        cout << "Fibonacci series: ";
        printfib(n);
    }
    return 0;
}
